### PR TITLE
core, verbs: Implement MR caching limitation with a total cached size

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
This PR adds the following updates to the core, utility, verbs code:
- core: New type of runtime params - `FI_PARAM_LONG`
- utility:
Implement MR caching limitation with a total cached size;
Change type of counters from `uint64_t` to `size_t` to align with their thresholds
- verbs:
Remove `FI_VERBS_MR_CACHE_LAZY_SIZE` variable
Rename `FI_VERBS_MR_CACHE_SIZE` to `FI_VERBS_MR_MAX_CACHED_CNT`
Add `FI_VERBS_MR_MAX_CACHED_SIZE` variable

### CODE REVIEW

File: prov/verbs/src/fi_verbs.c [hunk 1]
```diff
@@ -59,6 +59,7 @@ struct fi_ibv_gl_data fi_ibv_gl_data = {
 	.iface			= NULL,
 	.mr_cache_enable	= 0,
 	.mr_max_cached_cnt	= 4096,
+	.mr_max_cached_size	= 4294967296, /* 4 Gb */
```
Comment (a-ilango) [inline prov/verbs/src/fi_verbs.c]: How about `(1L << 32)`?

File: prov/verbs/src/fi_verbs.c [hunk 2]
```diff
@@ -465,6 +469,25 @@ static int fi_ibv_get_param_bool(const char *param_name,
...
+				  param_default);
+	if (ret)
+		return ret;
+
+	if (!fi_param_get_long(&fi_ibv_prov, param_name, &param))
+		*param_default = param;
+
+	return 0;
+}
+
```
Comment (a-ilango) [inline prov/verbs/src/fi_verbs.c]: In order to keep changes to the core in a separate patch, verbs specific changes can be moved to the patch that adds `FI_VERBS_MR_MAX_CACHED_SIZE ` env variable

File: include/ofi_mr.h [hunk 3]
```diff
@@ -185,12 +185,14 @@ struct ofi_mr_cache {
 	struct util_domain		*domain;
 	struct ofi_notification_queue	nq;
 	size_t				max_cached_cnt;
+	size_t				max_cached_size;
```
Comment (shefty) [inline include/ofi_mr.h]: Let's set this to a default of SIZE_MAX.  Then we can drop the checks to see if it's non-zero.

**Comment by a-ilango** [comment]
I think the patch that renames size to max_cached_cnt and the one that renames the corresponding env variable can go together in single patch.

File: prov/verbs/src/fi_verbs.c (hunk 1 above)
Comment (shefty) [reply prov/verbs/src/fi_verbs.c]: If this is size_t, then SIZE_MAX

Comment (gladkovdmitry17) [reply prov/verbs/src/fi_verbs.c]: Thanks, folks! Nice point

File: prov/verbs/src/fi_verbs.c (hunk 2 above)
Comment (gladkovdmitry17) [reply prov/verbs/src/fi_verbs.c]: I'e moved these changes into a separate patch

**Comment by gladkovdmitry17** [comment]
> I think the patch that renames size to max_cached_cnt and the one that renames the corresponding env variable can go together in single patch.

Yes, this makes sense to me

File: include/ofi_mr.h (hunk 3 above)
Comment (gladkovdmitry17) [reply include/ofi_mr.h]: Ditto

**Comment by gladkovdmitry17** [comment]
I'm merging this PR.
This functionality affects only verbs provider (+ RxM) as this uses MR caching and caching limited by size (implemented withing this PR) -- this isn't default behavior of verbs provider
